apiary.apib

```diff
@@ -0,0 +1,333 @@
+FORMAT: 1A
+HOST: http://lore.mit.edu/
+
+# LORE
+
+Simple API on top of MIT LORE application
+
+# URL
+URL Structure is `https://{domain}/api/v1/{resource}/{resourceId}`
+
+| Code   | Description |
+| ---    | --- |
+|`{domain}`    | The domain of the LORE service. `private-b0103d-lore.apiary-mock.com/` is the Apiary mock server. |
+|`/api/v1/`    | The service endpoint includes the current version (v1) |
+|`{resource}`  | The resource endpoint for a specific set of objects in the system. Supported endpoints include repositories, vocabularies, learningresources, members|
+|`{resource_idd}`  |The `resource_id` sets the ID for a specific object to reference |
+
+*Example: `http://lore.mit.edu/api/v1/repos` will return a json representation of all repositories.*
+
+*Example: `http://lore.mit.edu/api/v1/repos/123` will return a json representation of the repository with id=123.*
+
+# Operations
+LORE API uses the following HTTP verbs
+
+| Verb   | Description |
+| ---    | --- |
+|`GET`   |Select one or more items. Success returns 200 status code with json result item(s) in body.|
+|`POST`  |Create a new item. Success returns 201 status code with newly created json item in body.|
+|`PUT`   |Update an item. Success returns 200 status code with updated json item in body.|
+|`DELETE`|Delete an item. Success returns 204 status code with no body.|
+
+# Status Codes
+The API responds with one of the following status codes.
+
+| Code| Description |
+| --- | --- |
+|`200`| Success. Request completed. |
+|`201`| Success. New resource created. |
+|`204`| Success. No content to return. Only the status code returns. |
+|`400`| Bad Request - The request could not be parsed. |
+|`401`| Unauthorized - user is not logged in, cannot access resource. |
+|`404`| Not Found - resource doesn't exist. |
+|`422`| Unprocessable Entity - The request could not be parsed due to validation errors. |
+|`500`| Server error. |
+
+
+# Authentication
+
+tbd.
+
+# Pagination
+
+Requests that return multiple items all accept a `?page` parameter
+
+Example: `http://lore.mit.edu/api/v1/repos?page=2`
+
+Page numbering is 1-based. Omitting the ?page parameter will return the first page.
+
+# LORE API Root [/]
+
+## Retrieve the Entry Point [GET]
+
++ Response 200 (application/json)
+
+        {
+            "repositories_url": "/repos"
```

Comment:
Providing links like this is a property of HATEOAS, is that what we're going for?